src/backend/distributed/utils/maintenanced.c

```diff
@@ -626,6 +623,11 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 			ProcessConfigFile(PGC_SIGHUP);
 		}
 	}
+
+	if (metadataSyncBgwHandle)
```

Comment:
do you know what happens if the maintenance daemon exits through an error?

Reply:
Instead of doing this, I changed start-up of maintenance daemon to send SIGTERM to metadata sync daemons of crashed maintenance daemons.

Removed this so tests verify that approach.